### PR TITLE
Android: Fix AppComponent reconstructed on split screen

### LINKED ISSUE #5068: Switching between regular and split view restarts the app
When opening the android app in split view and making it fullscreen or the other way around the app restarts, losing which thread I was reading. This is very annoying when I want to switch away from split view to be easier to read a thread as I will have to navigate back to it. Other apps simply resize their view when switching without losing ui state.

### PR DESCRIPTION
**Pull request by Somena1**
This commit fixes #5068
Switching between screen size or split screen causes Android AppComponent to be reconstructed and I'm initiate a restart.

### CODE REVIEW

**Comment by chrisbobbe** [comment]
Thanks, @Somena1! I see two commits:

c4a976060cb9f023727875783672368b7f752bf8 (Android) Fix AppComponent reconstructed on split screen #5068
20648f1310713b52c05945785d88f28bfb981947 Android config changes set to alphabetical

It looks like the second commit is fixing up a small problem that was introduced in the first commit. Please squash the two commits together, so there's just one commit that doesn't (knowingly) introduce any problems.

For that single commit, please make sure it follows our commit-message style.

- The summary line shouldn't start with "(Android)". We have guidelines on how to write the prefix in the summary line: https://zulip.readthedocs.io/en/latest/contributing/version-control.html#good-summaries
- The commit-message body should end with `Fixes: #5068`: https://zulip.readthedocs.io/en/latest/contributing/version-control.html#message-body

Thanks!

**Comment by Somena1** [comment]
Hello @gnprice, I've resolved the squash and commit message. 
Thanks for putting me through

**Comment by Somena1** [comment]
Not sure if this is the right place to say this, my PR on RN addressing this issue was accepted 
[commit](https://github.com/facebook/react-native/commit/fc962c9)

**Comment by Somena1** [comment]
Hello @gnprice please review my changes

**Comment by Somena1** [comment]
@chrisbobbe please review changes in commit

**Comment by gnprice** [comment]
> Not sure if this is the right place to say this, my PR on RN addressing this issue was accepted [commit](https://github.com/facebook/react-native/commit/fc962c9)

Cool! Thanks for getting that fixed upstream.

No need to @-mention maintainers pinging for a review. It can take a few business days before we review a change, but we do see them.

For cross-reference, this PR is the continuation of #5073.

This change looks good! Thanks for your work on it. Merging.

**Comment by gnprice** [comment]
And done. I also revised the commit message -- take a look:

    android: Fix restart on resizing or splitting screen
    
    This fixes an issue where the app would restart on entering split
    screen, or on resizing the app's side of a split screen.
    
    Basically this line of the Android manifest is making an assurance to
    the system that the app is prepared to handle these changes; docs here:
      https://developer.android.com/guide/topics/manifest/activity-element#config
    When the system doesn't get that assurance, it just restarts the app
    from scratch to be sure.
    
    And empirically the app indeed does just fine when we add those
    assurances and then go ahead and switch to split-screen mode, or
    resize the app's portion of the split screen.  That layer of things is
    basically all RN's job, so RN is handling this just fine -- we just
    haven't been letting the system know that we're ready to do so.
    
    Also got this fixed in the RN template app upstream:
      https://github.com/facebook/react-native/pull/32536
      https://github.com/facebook/react-native/commit/fc962c9b6
    
    [greg: expanded commit message]
    
    Fixes: #5068

Thanks again!
